.gitignore

```diff
@@ -1,7 +1,43 @@
+# Ignore certain files #
+
+# CMake - General Files #
+
+cmake/
+cmakefiles/
+*.cmake
+CMakeCache.txt
```

Comment:
These should only be part of the build folder. Naming a folder for cmake modules or scripts is a very typical setup so I don't recommend ignoring that folder either